### DISCUSSION TITLE
Storage: Need a type definition for data.Key?

### DISCUSSION THREAD

**Discussion by ru88s** [supabase/supabase#1164] [Questions] [answered]
Need a type definition for data.Key?
What should i do?
I need directory path.
To create a public URL after uploading an image.

<img width="614" alt="スクリーンショット 2021-04-15 22 01 41" src="https://user-images.githubusercontent.com/42270511/114873362-2b7d8100-9e36-11eb-8b8b-ceb03eb9acb6.png">

```
 // imageUpload
const { data } = await supabase.storage.from("images").upload(`public/${uploadFile.name}`, uploadFile);
const imagePath = data.Key.replace("xxxx/images/", "");
```

**Comment by inian** [comment, marked as answer]
Good catch again @ru88s. 

I have made two changes in the recent supabase-js
- Fixed types for upload
- The returned value for `upload` and `update` doesn't return the project reference (the xxx in your code)

**Reply by ru88s** [reply]
Thank you again @inian.

>The returned value for upload and update doesn't return the project reference

Will xxx in my code disappear correctly?
My code is still returned.
I uploaded it.

![スクリーンショット 2021-04-15 22 52 20](https://user-images.githubusercontent.com/42270511/114880691-3e478400-9e3d-11eb-9a25-cab907937fa2.png)

**Reply by inian** [reply]
The server still returns it (will stop returning it in a day or two once we roll out the update). But I have already updated supabase-js to not return it. So if you use supabase-js, that xxx should disappear. Let me know if you see something else!

**Reply by ru88s** [reply]
I fully understand!
It will be very easy to use!
Thank you for your prompt response!
